internal/cfgplugins/mpls.go

```diff
@@ -389,3 +389,30 @@ func NewMPLSSRBasic(t *testing.T, batch *gnmi.SetBatch, dut *ondatra.DUTDevice,
 		gnmi.BatchReplace(batch, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).SegmentRouting().Config(), sr)
 	}
 }
+
+// LabelRangeOCConfig configures MPLS label ranges on the DUT using OpenConfig.
+func LabelRangeOCConfig(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Helper()
+	d := &oc.Root{}
+	ni := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	mplsObj := ni.GetOrCreateMpls().GetOrCreateGlobal()
+	// Map of local-id → [lowerBound, upperBound]
+	labelRanges := map[string][2]uint32{
+		"bgp-sr":                  {16, 0},
+		"dynamic":                 {16, 0},
+		"isis-sr":                 {16, 0},
+		"l2evpn":                  {16, 0},
+		"l2evpn ethernet-segment": {16, 0},
+		"ospf-sr":                 {16, 0},
+		"srlb":                    {16, 0},
+		"static":                  {16, 1048560},
+	}
+	t.Logf("Mpls Object %v, label range %v", mplsObj, labelRanges)
+	// TODO: The below loop will uncomment once OC paths are in place.
+	// for localID, bounds := range labelRanges {
+	// 	rlb := mplsObj.GetOrCreateReservedLabelBlock(localID)
+	// 	rlb.Config.LocalId = ygot.String(localID)
+	// 	rlb.Config.LowerBound = ygot.Uint32(bounds[0])
+	// 	rlb.Config.UpperBound = ygot.Uint32(bounds[1])
+	// }
```

Comment:
I don't understand?  This OC should be generated.  If a specific vendor doesn't support it, then a deviation may be proposed.    Since many other tests create label blocks, I suspect this issue has already been encountered.  For example, this test creates MPLS label blocks:

https://github.com/openconfig/featureprofiles/tree/3bac9da6be2973963744b812da946284b19d7201/feature/mpls/otg_tests/label_block

(this earlier test should have used cfgplugins so this would already be clear; thank you for getting it done.)

Reply:
Thanks for the reference. Updated code but still the label ranges are not supported via OpenConfig, so here required deviation.